vector.go

```diff
@@ -26,88 +25,71 @@ type vector struct {
 	getFn fnGetVectorValue
 	// A callback function to write to this vector.
 	setFn fnSetVectorValue
-	// The data type of the vector.
-	duckdbType C.duckdb_type
 	// The child vectors of nested data types.
 	childVectors []vector
 
-	// The child names of STRUCT vectors.
-	childNames []string
-	// The dictionary for ENUM types.
-	dict map[string]uint32
-	// The  width of DECIMAL types.
-	width uint8
-	// The scale of DECIMAL types.
-	scale uint8
```

Comment:
Nice, these always seems unnecessary to me :)

Reply:
Yes, I agree. Sadly, we do need some of them in the setters and getters. But at least they are in their dedicated `struct` now.